src/network/linkers_socket.cpp

```diff
@@ -186,7 +187,8 @@ void Linkers::Construct() {
   listener_->SetTimeout(socket_timeout_);
   listener_->Listen(incoming_cnt);
   std::thread listen_thread(&Linkers::ListenThread, this, incoming_cnt);
-  const int connect_fail_retry_cnt = 20;
+  const int connect_fail_retries_scale_factor = static_cast<int>(num_machines_ / 20);
+  const int connect_fail_retry_cnt = std::max(20, connect_fail_retries_scale_factor);
```

Comment:
@aakarshg Thank you for addressing comments! I'm not sure but maybe it will be better to store `20` in a constant? If we will need to update this value, that constant will allow us to do it in only one place.

Reply:
ping @aakarshg

Reply:
gently ping @aakarshg

Reply:
yes that'll be absolutely okay, i'll update the pr :) and apologies for late replies.

Reply:
Thank you!